src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -180,6 +185,7 @@ object ScalafixPlugin extends AutoPlugin {
 
   override lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
     scalafixConfig := None, // let scalafix-cli try to infer $CWD/.scalafix.conf
+    scalafixConfigOnCompile := scalafixConfig.value,
```

Comment:
I am not sure this is doing what we want as _I think_ we won't capture the `scalafixConfig` values set at `ThisBuild` or project level. `scalafixConfigOnCompile` could remain undefined, and the fallback happens at evaluation time, like https://github.com/scalacenter/sbt-scalafix/pull/140/files#diff-b6db8e3499369917214568aee2dd0960R357. In that case, you wouldn't need the `Option` wrapper on the setting key type.

Reply:
Thank you for pointing out that scalafixConfig in ThisBuild or project will not be captured (my bad).

And setting it as undefined and using `or` operator sounds better. I'll update it with tests.

Reply:
Since `or` operator doesn't work due to type parameter compatibility (SettingKey[File] v.s. SettingKey[Option[File]]) between scalafixOnCompileConfig and scalafixConfig, I'd use `?` operator for fallback implementation.

https://github.com/taisukeoe/sbt-scalafix/blob/92edd336cc87450ebc95d29e49684b8521ee89b2/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L355-L359